cmd/nerdctl/image_remove_linux_test.go

```diff
@@ -27,27 +27,66 @@ func TestRemoveImage(t *testing.T) {
 	tID := testutil.Identifier(t)
 	base.Cmd("image", "prune", "--force", "--all").AssertOK()
 
-	// ignore error
-	base.Cmd("rmi", "-f", tID).AssertOK()
-
 	base.Cmd("run", "--name", tID, testutil.CommonImage).AssertOK()
-	defer base.Cmd("rm", "-f", tID).Run()
+	defer base.Cmd("rm", "-f", tID).AssertOK()
 
 	base.Cmd("rmi", testutil.CommonImage).AssertFail()
-	defer base.Cmd("rmi", "-f", testutil.CommonImage).Run()
 	base.Cmd("rmi", "-f", testutil.CommonImage).AssertOK()
-
-	base.Cmd("images").AssertNoOut(testutil.CommonImage)
+	base.Cmd("images").AssertNoOut(testutil.ImageRepo(testutil.CommonImage))
 }
 
 func TestRemoveRunningImage(t *testing.T) {
 	base := testutil.NewBase(t)
 	tID := testutil.Identifier(t)
+
+	base.Cmd("run", "--name", tID, "-d", testutil.CommonImage, "sleep", "infinity").AssertOK()
+	defer base.Cmd("rm", "-f", tID).AssertOK()
+
+	base.Cmd("rmi", testutil.CommonImage).AssertFail()
+
+	// `rmi -f` only logs err, not return err. need to check `images` output
+	base.Cmd("kill", tID).AssertOK()
+	base.Cmd("rmi", testutil.CommonImage).AssertFail()
+	base.Cmd("rmi", "-f", testutil.CommonImage).AssertOK()
+	base.Cmd("images").AssertNoOut(testutil.ImageRepo(testutil.CommonImage))
+}
+
+func TestRemovePausedImage(t *testing.T) {
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+
 	base.Cmd("run", "--name", tID, "-d", testutil.CommonImage, "sleep", "infinity").AssertOK()
-	defer base.Cmd("rm", "-f", tID).Run()
+	base.Cmd("pause", tID).AssertOK()
+	defer base.Cmd("rm", "-f", tID).AssertOK()
+
+	// `rmi -f` cannot remove image (`-f` only logs err, not return err)
 	base.Cmd("rmi", testutil.CommonImage).AssertFail()
+	base.Cmd("rmi", "-f", testutil.CommonImage).AssertOK()
+	base.Cmd("images").AssertOutContains(testutil.ImageRepo(testutil.CommonImage))
+
+	// `rmi -f` can remove image after container killed
 	base.Cmd("kill", tID).AssertOK()
 	base.Cmd("rmi", testutil.CommonImage).AssertFail()
 	base.Cmd("rmi", "-f", testutil.CommonImage).AssertOK()
-	base.Cmd("images").AssertNoOut(testutil.CommonImage)
+	base.Cmd("images").AssertNoOut(testutil.ImageRepo(testutil.CommonImage))
+}
+
+func TestRemoveImageWithCreatedContainer(t *testing.T) {
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+
+	base.Cmd("pull", testutil.AlpineImage).AssertOK()
+	base.Cmd("pull", testutil.NginxAlpineImage).AssertOK()
+
+	base.Cmd("create", "--name", tID, testutil.AlpineImage, "sleep", "infinity").AssertOK()
+	defer base.Cmd("rm", "-f", tID).AssertOK()
+
+	// `rmi -f` can remove image associated with created container (`-f` only logs err, not return err)
+	base.Cmd("rmi", testutil.AlpineImage).AssertFail()
+	base.Cmd("rmi", "-f", testutil.AlpineImage).AssertOK()
+	base.Cmd("images").AssertNoOut(testutil.ImageRepo(testutil.AlpineImage))
+
+	// a created container with removed image doesn't impact other `rmi` command
+	base.Cmd("rmi", "-f", testutil.NginxAlpineImage).AssertOK()
```

Comment:
Can this work without force rmi with a retagged image name?

Reply:
replied in comment above.